### PR TITLE
Added itemSaverThreshold setting

### PR DESCRIPTION
**Pull request by mankool0**
This setting allows you to change the durability at which baritone should stop using a tool when using the itemSaver setting. Currently baritone will stop using the tool when it's at 0 durability which sometimes breaks it. I've set the default value to 10 to give a little buffer to avoid tools breaking.

### CODE REVIEW

File: src/api/java/baritone/api/Settings.java [hunk 1]
```diff
@@ -733,6 +733,11 @@
      */
     public final Setting<Boolean> itemSaver = new Setting<>(false);
 
+    /**
+     * Durability to leave on the tool when using itemSaver
+     */
+    public final Setting<Integer> itemSaverThreshold = new Setting<>(10);
```
Comment (ZacSharp) [inline src/api/java/baritone/api/Settings.java:739]: Ideally this would be one setting (-1 ≙ itemSaver off) but it's a boolean already and I don't think you should change that.
TLDR: imo you've don it right

**Review by ZacSharp — APPROVED** [review]
